samples/Mvc.FileUpload/Controllers/FileController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+using Mvc.FileUpload.Filters;
+
+namespace Mvc.FileUpload.Controllers
+{
+    public class FileController : Controller
+    {
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly ILogger<FileController> _logger;
+
+        // Get the default form options so that we can use them to set the default limits for
+        // request body data
+        private static readonly FormOptions _defaultFormOptions = new FormOptions();
+
+        public FileController(IHostingEnvironment hostingEnvironment, ILogger<FileController> logger)
+        {
+            _hostingEnvironment = hostingEnvironment;
+            _logger = logger;
+        }
+
+        [HttpGet]
+        [GenerateAntiforgeryTokenCookieForAjax]
+        public IActionResult FormToUpload()
+        {
+            return View();
+        }
+
+        // 1. We disable the form value model binding here to take control of handling potentially large files.
+        // 2. Typically antiforgery tokens are sent in request body, but since we do not want to read the request body
+        //    early, the tokens are made to be sent via headers. The antiforgery token filter first looks for tokens
+        //    in the request header and then falls back to reading the body.
+        [HttpPost]
+        [DisableFormValueModelBinding]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Upload()
+        {
+            if (!MultipartRequestHelper.IsMultipartContentType(Request.ContentType))
+            {
+                return BadRequest($"Expected a multipart request, but got '{Request.ContentType}'.");
+            }
+
+            //TODO: HOW CAN I MODEL BIND ALL THIS FORM URL ENCODED DATA LATER
+
+            // Used to accumulate all the form url encoded key value pairs in the request.
+            var formAccumulator = new KeyValueAccumulator();
```

Comment:
Nothing in the sample consumes the accumulated values. Should there be a success view showing the fields and file stats?